FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90

```diff
@@ -5360,11 +5360,15 @@ subroutine GFS_physics_driver                                 &
               Statein%prsi(1:im,1:levs+1), im, levs, nwat, 1, Model%ntcw, Model%ntiw, &
               Model%ntrw, Model%ntsw, Model%ntgl, specific_heat)
         endif
-
-        call update_temperature_tendency_diagnostics(Diag%t_dt, dt3dt_initial, Diag%dt3dt, &
-            specific_heat, im, levs, dtp)
-        call update_water_vapor_tendency_diagnostics(Diag%q_dt, dq3dt_initial, Diag%dq3dt, &
-            Statein%qgrs(:,:,1:nwat), Stateout%gq0(:,:,1:nwat), im, levs, nwat, dtp)
+        call compute_updated_delp_following_dynamics_definition(Statein%prsi(1:im,1:levs+1), &
```

Comment:
so this will always happen now. this doesn't change what the dycore sees/uses for delp? probably not otherwise you would have had to change the model regressions

Reply:
It will happen if `gfs_physics_nml.ldiag3d` is set to `.true.` in the namelist (by default it is `.false.`).  For context, `ldiag3d` controls whether memory is allocated to compute 3D physics diagnostics.  That's right, though, as such this is purely diagnostic; it does not change how `delp` is updated in the dynamics.